allennlp/training/trainer.py

```diff
@@ -290,14 +305,14 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
         # Set the model to "train" mode.
         self.model.train()
 
-        num_gpus = len(self._cuda_devices)
+        num_batch_groups = self._num_gradient_accumulation_steps * len(self._cuda_devices)
```

Comment:
can you add a comment explaining what `num_batch_groups` is and why it's computed the way it is, it's not obvious to me

Reply:
I realized that `num_batch_groups` is a misnomer. It denotes the length of each `batch_group` that gets into the training loop. I've renamed it as `batch_group_length`. This is just an extension of the flow that is being used for the `DataParallel` case. I've added the explanation as comments.